django/core/management/commands/runserver.py

```diff
@@ -134,6 +135,9 @@ def inner_run(self, *args, **options):
         # Need to check migrations here, so can't use the
         # requires_migrations_check attribute.
         self.check_migrations()
+        # Close all connections opened during migration checking.
+        for conn in connections.all(initialized_only=True):
```

Comment:
Technically this doesn't have to be in this PR and can go in before in a separate one if you prefer.